config/config-api/src/com/thoughtworks/go/util/command/EnvironmentVariableContext.java

```diff
@@ -227,25 +224,21 @@ public int hashCode() {
         return properties != null ? properties.hashCode() : 0;
     }
 
-    public String reportText() {
-        throw new RuntimeException("IMPLEMENT ME");
-    }
-
-    public void setupRuntimeEnvironment(Map<String, String> env, ConsoleOutputStreamConsumer consumer) {
+    public List<String> report() {
```

Comment:
I mean, there's a difference between `env.containsKey` and `names.contains`. Particularly because of [this line](https://github.com/gocd/gocd/blob/2df65395774dd0ed4f3b2dc0df1bf3043d675925/common/src/com/thoughtworks/go/util/ProcessManager.java#L60).